src/tests/providers/test_cudo.py

```diff
@@ -0,0 +1,131 @@
+from typing import List
+
+import pytest
+
+from gpuhunt import QueryFilter
+from gpuhunt.providers.cudo import (
+    CudoProvider,
+    get_balanced_disk_size,
+    get_balanced_memory,
+    get_memory,
+    gpu_name,
+    optimize_offers_with_gpu,
+)
+
+
+@pytest.fixture
+def machine_types() -> List[dict]:
+    return [
+        {
+            "dataCenterId": "br-saopaulo-1",
+            "machineType": "cascade-lake",
+            "cpuModel": "Cascadelake-Server-noTSX",
+            "gpuModel": "RTX 3080",
+            "gpuModelId": "nvidia-rtx-3080",
+            "minVcpuPerMemoryGib": 0.25,
+            "maxVcpuPerMemoryGib": 1,
+            "minVcpuPerGpu": 1,
+            "maxVcpuPerGpu": 13,
+            "vcpuPriceHr": {"value": "0.002500"},
+            "memoryGibPriceHr": {"value": "0.003800"},
+            "gpuPriceHr": {"value": "0.05"},
+            "minStorageGibPriceHr": {"value": "0.00013"},
+            "ipv4PriceHr": {"value": "0.005500"},
+            "maxVcpuFree": 76,
+            "totalVcpuFree": 377,
+            "maxMemoryGibFree": 227,
+            "totalMemoryGibFree": 1132,
+            "maxGpuFree": 5,
+            "totalGpuFree": 24,
+            "maxStorageGibFree": 42420,
+            "totalStorageGibFree": 42420,
+        }
+    ]
+
+
+def test_get_offers_with_query_filter():
+    cudo = CudoProvider()
+    offers = cudo.get(QueryFilter(min_gpu_count=1, max_gpu_count=1), balance_resources=True)
+    print(f"{len(offers)} offers found")
```

Comment:
Use the logger instead of the print function. The logger can be configured during the tests.

Reply:
Removed print and improved test with mocks.

Reply:
Thank you!